src/XmlScope.ts

```diff
@@ -153,6 +153,46 @@ export class XmlScope extends Scope {
         });
     }
 
+    public getCodeActions(file: BscFile, range: Range) {
+        const result = [] as CodeAction[];
+        for (const diagnostic of this.diagnostics) {
```

Comment:
This filtering of the diagnostics should be done once at the reception of the code actions, otherwise you'll end up with duplicated logic of looking up for diagnostics matching the range.

Reply:
Done.